main.go

```diff
@@ -17,17 +17,23 @@ limitations under the License.
 package main
 
 import (
+	"context"
 	"flag"
+	"net/http"
 	"os"
 	"os/signal"
 	"sync"
 	"syscall"
 	"time"
 
+	"github.com/0xRLG/ocworkqueue"
```

Comment:
Is there a benefit to going via OpenCensus instead of reusing the [Prometheus provider](https://sourcegraph.com/github.com/kubernetes/kubernetes/-/blob/pkg/util/workqueue/prometheus/prometheus.go) from core k8s? Whenever possible, I prefer to reuse code and patterns from the core since they're well-exercised.

Reply:
I didn't even know that existed I've been using this package for a long time now and I've only just made it open source. There isn't a huge difference between the two plain old prometheus and opencensus. Both result in Prometheus metrics being exposed on the debug endpoint. I think that OpenCensus ends up being cheaper to collect if you don't report it anywhere but we don't do that in this PR.

To be honest I think the fact that this project is in GoogleCloudPlatform and OpenCensus is being driven by Google people the two together made sense to me.

Reply:
Alright that makes sense. Let's just keep OpenCensus. This PR should be good to go then whenever you have a chance to make the other changes below.